python_modules/dagster-externals/dagster_externals/_io/base.py

```diff
@@ -37,3 +42,58 @@ def load_context_params(self) -> ExternalExecutionParams:
     @abstractmethod
     def load_messages_params(self) -> ExternalExecutionParams:
         ...
+
+
+class ExternalExecutionBlobStoreMessageWriter(ExternalExecutionMessageWriter):
+    interval: float
+    max_chunk_size: int
+    buffer: List[ExternalExecutionMessage]
+    counter: int
+
+    def __init__(self, *, interval: float = 10):
+        self.interval = interval
+        self.buffer = []
+        self.counter = 1
+
+    @contextmanager
+    def setup(self, params: ExternalExecutionParams) -> Iterator[None]:
+        self.set_params(params)
+        thread = None
+        is_task_complete = Event()
+        try:
+            thread = Thread(target=self.writer_thread, args=(is_task_complete,), daemon=True)
+            thread.start()
+            yield
+        finally:
+            is_task_complete.set()
+            if thread:
+                thread.join()
```

Comment:
backstop `join` with a timeout so it doesn't block indefinitely in edge case / future bug